sunbeam-python/sunbeam/features/maintenance/checks.py

```diff
@@ -271,3 +291,339 @@ def run(self) -> bool:
+                self.deployment.get_client(),
+                self.jhelper,
+                self.deployment.openstack_machines_model,
+                node,
+                K8S_APP_NAME,
+            )
+        except UnitNotFoundException:
+            LOG.debug("cannot find %s unit in '%s'", K8S_APP_NAME, node)
+            return ""
+
+    def _has_active_k8s_dqlite_svc(self, k8s_unit_name: str) -> bool:
+        """Check if the k8s unit has active k8s dqlite svc."""
+        try:
+            result = self.jhelper.run_cmd_on_machine_unit_payload(
+                k8s_unit_name,
+                self.deployment.openstack_machines_model,
+                f"snap services {K8S_DQLITE_SVC_NAME} | grep active",
+                timeout=COMMAND_TIMEOUT,
+            )
+            LOG.debug(result)
+            status = result.stdout.split()[2]
+        except (ExecFailedException, IndexError) as e:
+            LOG.debug("%s", str(e))
+            return False
+        return "active" == status
+
+
+class NoJujuControllerPodCheck(Check):
+    """Check if the node has juju controller pods."""
+
+    def __init__(self, node: str, deployment: Deployment):
+        super().__init__(
+            "Check if there are juju contoller pods in the node.",
+            "Checking if there are juju contoller pods in the node.",
+        )
+        self.node = node
+        self.deployment = deployment
+
+    def run(self) -> bool:
+        """Check if the node has juju controller pods."""
+        if is_maas_deployment(self.deployment):
+            LOG.debug(
+                "Skipped checking juju controller pods. MAAS deployment have"
+                " juju-controller role instead of controle role."
+            )
+            return True
+
+        juju_controller = self.deployment.juju_controller
+        if juju_controller and juju_controller.is_external:
+            LOG.debug(
+                "Skipped checking juju controller pods. Juju controller pod is not"
+                " managed by Sunbeam for manual bare metal provider with an external."
+            )
+            return True
+
+        try:
+            juju_controller_pods = self._fetch_juju_controller_pods()
+        except KubeClientError:
+            self.message = "Failed to get k8s client"
+            return False
+
+        has_juju_controller_pods = False
+        for pod in juju_controller_pods:
+            # Found juju controller pod in this node
+            if pod.spec and pod.spec.nodeName == self.node:
+                LOG.debug(
+                    "Found juju controller pod '%s' in '%s'",
+                    pod.metadata.name,
+                    self.node,
+                )
+                has_juju_controller_pods = True
+
+        if not has_juju_controller_pods:
+            return True
+
+        self.message = (
+            f"cannot enable maintenance mode for '{self.node}' because this"
+            " node hosts juju controller pods."
```

Comment:
It will be great to provide and explanation on why maintenance mode cannot be enabled if juju controller pod exists on the node. OR an explanation in documentation page and refer the link here in the message (can be done as separate PR once the documentation is merged)

Reply:
My plan is the have the documentation in https://canonical-openstack.readthedocs-hosted.com/en/latest/explanation/maintenance-mode/. And this page will explain what are the preflight checks, and how can users address them here. Do you suggest having a different page dedicated to juju controller pod? (my plan is to include the result in that doc as well)